README.md
# publisher

## Project setup

```bash
yarn install
```

### Compiles and hot-reloads for development

Rename or copy the env/.env.example env/.env.local

fill in the configuration

and then run

```bash
yarn electron:serve
```

### Compiles and minifies for production

Rename or copy the env/.env.example env/.env

fill in the configuration

and then run

```bash
yarn electron:build
```

or

```bash
yarn electron:win-build
```

### Run your unit tests

```bash
yarn test:unit
```

### Lints and fixes files

```bash
yarn lint
```

### Customize configuration

See [Configuration Reference](https://cli.vuejs.org/config/).


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [0.9.21](https://github.com/Laev/Publisher/compare/v0.9.20...v0.9.21) (2022-09-30)


### Features

* **auth:** update security strategy ([6e67741](https://github.com/Laev/Publisher/commit/6e67741036bbd60db968ea794038ecfd58ec342d))

### [0.9.20](https://github.com/Laev/Publisher/compare/v0.9.19...v0.9.20) (2022-08-31)

### [0.9.19](https://github.com/Laev/Publisher/compare/v0.9.18...v0.9.19) (2022-04-19)

### [0.9.18](https://github.com/Laev/Publisher/compare/v0.9.17...v0.9.18) (2022-04-12)

### [0.9.17](https://github.com/Laev/Publisher/compare/v0.9.16...v0.9.17) (2022-04-12)

### [0.9.16](https://github.com/Laev/Publisher/compare/v0.9.15...v0.9.16) (2022-04-11)

### [0.9.15](https://github.com/Laev/Publisher/compare/v0.9.14...v0.9.15) (2022-04-11)

### [0.9.14](https://github.com/Laev/Publisher/compare/v0.9.13...v0.9.14) (2022-04-11)


### Bug Fixes

* **workflow:** config update ([65a7f2e](https://github.com/Laev/Publisher/commit/65a7f2ef06f82d3895a643340ae03780f146c124))

### [0.9.13](https://github.com/Laev/Publisher/compare/v0.9.12...v0.9.13) (2022-04-11)

### [0.9.12](https://github.com/Laev/Publisher/compare/v0.9.11...v0.9.12) (2022-04-11)

### [0.9.11](https://github.com/Laev/Publisher/compare/v0.9.10...v0.9.11) (2022-04-11)

### [0.9.10](https://github.com/Laev/Publisher/compare/v0.9.9...v0.9.10) (2022-04-11)

### [0.9.9](https://github.com/Laev/Publisher/compare/v0.9.8...v0.9.9) (2022-04-11)


### Bug Fixes

* **releasetype:** change releaseType ([8caab32](https://github.com/Laev/Publisher/commit/8caab32e34f3150411a33eb5676d60ea19a54903))

### [0.9.8](https://github.com/Laev/Publisher/compare/v0.9.7...v0.9.8) (2022-04-10)


### Bug Fixes

* **lockfile:** yarn lockfile fix ([00f7b89](https://github.com/Laev/Publisher/commit/00f7b896ddcc02b8387630d117b880ba3f871c26))

### [0.9.7](https://github.com/Laev/Publisher/compare/v0.9.6...v0.9.7) (2022-04-10)


### Bug Fixes

* **workflow:** config update ([93d5aaf](https://github.com/Laev/Publisher/commit/93d5aaf70103f60cb01b1b3e28b10a8f8240966c))

### [0.9.6](https://github.com/Laev/Publisher/compare/v0.9.5...v0.9.6) (2022-04-10)


### Bug Fixes

* **lockfile:** update lock file ([b34b1f9](https://github.com/Laev/Publisher/commit/b34b1f9e47d16cccff68a6206e969745666348bb))

### [0.9.5](https://github.com/Laev/Publisher/compare/v0.9.4...v0.9.5) (2022-04-10)


### Bug Fixes

* **lockfile:** remove yarn lock file ([ec69261](https://github.com/Laev/Publisher/commit/ec692614f5f4daef46c7153f13ecca7040a8531e))

### [0.9.4](https://github.com/Laev/Publisher/compare/v0.9.3...v0.9.4) (2022-04-10)


### Bug Fixes

* **workflow:** yml update ([0548511](https://github.com/Laev/Publisher/commit/0548511adac371d2de35415eb684c7edbb4afd40))

### [0.9.3](https://github.com/Laev/Publisher/compare/v0.9.2...v0.9.3) (2022-04-08)


### Bug Fixes

* **workflow:** update yml ([042f416](https://github.com/Laev/Publisher/commit/042f4164891148947a851d1709031e212b8932a8))

### [0.9.2](https://github.com/Laev/Publisher/compare/v0.9.1...v0.9.2) (2022-04-08)


### Bug Fixes

* **release:** try to build ([c2a44fa](https://github.com/Laev/Publisher/commit/c2a44fa445eac54e2040cdda0ba60eda0a445c49))

### [0.9.1](https://github.com/Laev/Publisher/compare/v0.9.0...v0.9.1) (2022-04-08)


### Bug Fixes

* **env:** in order to eliminate GitHub warning to delete false environment variables value ([860a063](https://github.com/Laev/Publisher/commit/860a0639ec27537fd8f05ce40520d7d708125e47))

## [0.9.0](https://github.com/Laev/Publisher/compare/v0.8.4-rc7...v0.9.0) (2022-04-08)


### Features

* **env:** 在配置文件中装载敏感信息 & 尝试使用github action ([365e97b](https://github.com/Laev/Publisher/commit/365e97be2b00199108db61474a965c22298fa2c5))

<!--
 * @Author: Whzcorcd
 * @Date: 2020-12-15 14:43:54
 * @LastEditors: Whzcorcd
 * @LastEditTime: 2021-01-29 18:12:29
 * @Description: file content
-->
# :tada: 0.7.8-rc6 (2021-01-29)

### :sparkles: Features

* 研发端-增加修改项目属性功能
* 研发端-优化时间段提醒功能
* 研发端-增加分包配置

### :bug: Bug Fixes

* 研发端-修复发布工作流中 gitlab 项目 id 选取问题



# :tada: 0.7.7-rc6 (2021-01-27)

### :bug: Bug Fixes

* 研发端-修复 npm-conf 依赖在 win32 环境下缓存地址读取为 undefined 的问题



# :tada: 0.7.6-rc6 (2021-01-13)

### :sparkles: Features

* 产品经理端-增加更新驳回功能
* 研发端-增加生产环境构建时间提醒
* 研发端-增加项目多选时的选择处理

### :bug: Bug Fixes

* 产品经理端-修复邮件 quoted-printable 编码时无法正常解析的问题
* 产品经理端-修复个人资料填写页路由跳转问题



# :tada: 0.7.5-rc6 (2021-01-11)

### :bug: Bug Fixes

* 修复研发端菜单消失问题
* 修复 iconv-lite 依赖 webpack.require(...) 报错的问题



# :tada: 0.7.4-rc6 (2021-01-11)

### :sparkles: Features

* 新增静态扫描工作流
* 优化推荐工作流流程
* 完善邮件更新机制，通过 id 比对识别已回复的邮件
* 增加邮件轮询和新邮件状态提示

### :bug: Bug Fixes

* 修复更新状态显示异常问题



# :tada: 0.7.3-rc6 (2021-01-08)

### :sparkles: Features

* 新增项目时自动绑定对应 Gitlab 仓库
* 构建后加入更新数据流程
* 工作流拥有推荐流程
* 集成 PM 账号体系
* 集成邮件收发功能，PM 可以通过 Publisher 实现一键回复验证



# :tada: 0.6.8-rc5 (2020-12-30)

### :sparkles: Features

* 增加任务执行前未合并 MR 检测机制
* 细节修改



# :tada: 0.6.7-rc5 (2020-12-30)

### :sparkles: Features

* 增加任务执行历史记录功能，并在错误时可以提供错误信息



# :tada: 0.6.6-rc5 (2020-12-30)

### :sparkles: Features

* 实现全局任务状态，并且由列表页局部状态同步全局状态

### :bug: Bug Fixes

* 修复任务状态录入 action 时存在的问题



# :tada: 0.6.5-rc5 (2020-12-29)

### :sparkles: Features

* 实现全局更新状态和任务状态
* 拆分数据存储模块，优化数据库读写，增加备份功能
* 增加对过期数据的校验，增加页面缓存

### :bug: Bug Fixes

* 修改工作流任务原有并行模式为串行模式
* 修复工作流模块中存在的已知问题



# :tada: 0.6.4-rc5 (2020-12-25)

### :bug: Bug Fixes

* 修复数据不兼容导致的问题



# :tada: 0.6.3-rc5 (2020-12-25)

### :sparkles: Features

* 拆分 task 模块为多文件分散组合，工作流底层实现重构，所有任务由完全独立改为相互依赖，并实现全局参数概念
* 优化 jenkins 模块逻辑，现在可以侦测 jenkins 构建过程，在构建没有完毕时任务不会提前结束
* 优化 notify 模块逻辑，引入 webhook 通知池概念，实现环境区分通知
* 实现参数按需录入，完善工作流执行概念
* 使用全新的项目信息，摒弃无用的数据

### :bug: Bug Fixes

* 修复工作流执行时可能导致的对象引用的问题
* 增加部分界面参数缺失时的判断



# :tada: 0.5.5-rc4 (2020-12-24)

### :bug: Bug Fixes

* 修复前后分支相同时意外产生的分支合并操作
* 替换使用更加安全的 id 生成器模块



# :tada: 0.5.4-rc4 (2020-12-23)


vue.config.js
/*
 * @Author: Whzcorcd
 * @Date: 2020-12-04 18:09:44
 * @LastEditors: Whzcorcd
 * @LastEditTime: 2021-07-25 22:50:31
 * @Description: file content
 */
const path = require('path')
const Dotenv = require('dotenv-webpack')

const plugins = []
if (!process.env.NODE_ENV || process.env.NODE_ENV === 'development') {
  plugins.push(new Dotenv({ expand: true, path: './env/.env.local' }))
} else {
  plugins.push(new Dotenv({ expand: true, path: './env/.env' }))
}

function resolve(dir) {
  return path.join(__dirname, dir)
}

module.exports = {
  productionSourceMap: false,
  chainWebpack: config => {
    config.resolve.alias.set('#', resolve('public/main'))
    config.resolve.alias.set('*', resolve('src/app'))
    config.optimization.splitChunks({
      chunks: 'all',
      minSize: 30000,
      maxAsyncRequests: Infinity,
      maxInitialRequests: 3,
      automaticNameDelimiter: '-',
      cacheGroups: {
        libs: {
          test: /[\\/]node_modules[\\/]/,
          name: 'chunk-libs',
          priority: 10,
          chunks: 'initial' // 只打包初始时依赖的第三方
        },
        elementUI: {
          test: /[\\/]node_modules[\\/]element-ui[\\/]/,
          name: 'chunk-elementUI', // 单独将 elementUI 拆包
          priority: 20 // 权重要大于 libs 和 app 不然会被打包进 libs 或者 app
        },
        commons: {
          name: 'chunk-commons',
          minChunks: 2, // 最小共用次数
          priority: 10
        }
      }
    })
  },
  configureWebpack: {
    // bug from 'iconv-lite', https://github.com/ashtuchkin/iconv-lite/issues/204
    module: {
      rules: [
        {
          test: /node_modules[\\/\\](iconv-lite)[\\/\\].+/,
          resolve: {
            aliasFields: ['main']
          }
        }
      ]
    },
    plugins
  },
  pluginOptions: {
    electronBuilder: {
      nodeIntegration: true,
      builderOptions: {
        appId: 'com.gdyfe',
        productName: 'Publisher',
        copyright: 'Copyright © 2022 Guangdianyun',
        directories: {
          output: './dist_electron' // 输出文件路径
        },
        publish: [
          {
            provider: 'github',
            owner: 'Laev',
            repo: 'Publisher',
            releaseType: 'draft'
          }
        ],
        snap: {
          publish: ['github']
        },
        win: {
          // windows 相关配置
          icon: 'build/icons/icon.ico',
          target: [
            {
              target: 'nsis',
              arch: [
                'x64' // 64位
              ]
            }
          ]
        },
        mac: {
          icon: 'build/icons/icon.icns',
          target: ['zip', 'pkg', 'dmg']
        },
        linux: {
          icon: 'build/icons/'
        },
        nsis: {
          shortcutName: 'Publisher',
          uninstallDisplayName: '卸载这个软件',
          oneClick: false,
          perMachine: false,
          allowToChangeInstallationDirectory: true,
          allowElevation: true,
          createDesktopShortcut: true
        }
      }
    }
  },
  css: {
    sourceMap: false,
    loaderOptions: {
      sass: {
        sourceMap: false,
        prependData: `@import "~@/assets/styles/element-variables.scss";`
      }
    }
  }
}
